src/ml_tooling/baseclass.py

```diff
@@ -139,20 +161,47 @@ def save_estimator(self, path: str) -> pathlib.Path:
                     "You haven't scored the estimator - no results available to log"
                 )
 
-            metric_scores = {self.result.metric: float(self.result.score)}
-
-            log_results(
-                metric_scores=metric_scores,
-                estimator_name=self.estimator_name,
-                estimator_params=self.result.estimator_params,
-                run_dir=self.config.RUN_DIR,
-                estimator_path=str(estimator_file),
-            )
+            log = Log.from_result(self.result, estimator_path=estimator_file)
+            log.save_log(self.config.RUN_DIR)
 
         logger.info(f"Saved estimator to {estimator_file}")
 
         return estimator_file
 
+    def dump(self):
```

Comment:
When we use dump I think one would expect it to be a pickle file. Maybe we should use another name.

Reply:
Is now `to_dict`